clipboard/ios/Plugin/Clipboard.swift

```diff
@@ -0,0 +1,78 @@
+import Foundation
+import Capacitor
+
+public class Clipboard: NSObject {
+
+    enum ContentType: Int {
+        case string
+        case url
+        case image
+    }
+
+    struct ClipboardError: Error {
+        private let message: String
+
+        var localizedDescription: String {
+            return message
+        }
+
+        init(_ message: String) {
+            self.message = message
+        }
+    }
```

Comment:
This is actually a minor anti-pattern because you're performing a type erasure on the error. The only thing that distinguishes one error from another is the message, which is private and therefore invisible. Unless you need to store extra fields of information (in which case a struct is appropriate), an enum is more expressive. Additionally, you're overwriting the existing implementation of `localizedDescription`. The proper way to supply custom values to that method is to conform to `LocalizedError` and overwrite `errorDescription`.
```suggestion
  enum ClipboardError: LocalizedError {
    case invalidURL, invalidImage
    
    public var errorDescription: String? {
        switch self {
        case .invalidURL:
            return "Unable to form URL"
        case .invalidImage:
            return "Unable to encode image"
        }
    }
}
```